versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapter.java

```diff
@@ -174,7 +182,11 @@ Hash transplant(
    *     branch due to a conflicting change or if the expected hash of {@code toBranch} is not its
    *     expected hEAD
    */
-  Hash merge(Hash from, BranchName toBranch, Optional<Hash> expectedHead)
+  Hash merge(
```

Comment:
I don't think its the responsibility of the caller to decide how commit metadata gets updated by a transplant or merge. I think we want to only update the commit time and committer when a commit has been transplanted so I think we really only need to pass down the principal

Reply:
By the existing design of this code, the lower layers of code treat metadata as a blackbox - For VersionStore, METADATA is generic, and for DBAdapter, metadata is a ByteString.

The main reason for the approach, where caller dictates the transformation logic, is because only the caller is aware of the `CommitMeta` object structure. 
The attempt here is to align with the current state, and not deviate the dependency graph across modules. Setting the principal in the VersionStore/DBAdapter would otherwise, be required to understand the model class - CommitMeta.

Reply:
thats valid. I am torn here on one hand the reason for updating the metadata on transplant/merge is to ensure strong auditing (same as only setting principal on the server side) on the other hand you are correct we don't want to expose internal info about commit meta to the commit kernel.

Maybe I would feel more comfortable with a method to modify commit meta that was only called by transplant/merge. I think it would be easier to audit and could have a big warning comment about what is allowed to be changed in that method. Thoughts?

Reply:
The method essentially feeds commit time properties in the meta object, this is applicable for all types of commits (merge/transplant, as well as the regular ones). May be we should rename the method to exactly say what it does - `setCommitProperties` with a big warning comment preventing people from not doing anything else in there. WDYT?